templates/types/streaming/nextjs/app/components/ui/upload-csv-preview.tsx

```diff
@@ -1,55 +1,88 @@
 import { XCircleIcon } from "lucide-react";
 import Image from "next/image";
 import SheetIcon from "../ui/icons/sheet.svg";
+import { Button } from "./button";
+import { CsvFile } from "./chat";
+import {
+  Drawer,
+  DrawerClose,
+  DrawerContent,
+  DrawerDescription,
+  DrawerHeader,
+  DrawerTitle,
+  DrawerTrigger,
+} from "./drawer";
 import { cn } from "./lib/utils";
 
-export default function UploadCsvPreview({
-  filename,
-  filesize,
-  onRemove,
-  isNew,
-}: {
-  filename: string;
-  filesize: number;
-  onRemove: () => void;
+export interface UploadCsvPreviewProps {
+  csv: CsvFile;
+  onRemove?: () => void;
   isNew?: boolean;
-}) {
+}
+
+export default function UploadCsvPreview(props: UploadCsvPreviewProps) {
+  const { filename, filesize, content } = props.csv;
+  const { onRemove, isNew } = props;
   const fileSizeInKB = Math.round((filesize / 1024) * 10) / 10;
   return (
-    <div className="p-2 w-60 max-w-60 bg-secondary rounded-lg text-sm relative">
-      <div className="flex flex-row items-center gap-2">
-        <div className="relative h-10 w-10 shrink-0 overflow-hidden rounded-md">
-          <Image
-            className="h-full w-auto"
-            priority
-            src={SheetIcon}
-            alt="SheetIcon"
-          />
-        </div>
-        <div className="overflow-hidden">
-          <div className="truncate font-semibold">
-            {filename} ({fileSizeInKB} KB)
+    <Drawer direction="left">
+      <DrawerTrigger asChild>
+        <div className="p-2 w-60 max-w-60 bg-secondary rounded-lg text-sm relative cursor-pointer">
```

Comment:
factor out the preview class (same file don't export)?

Reply:
no need to detach, I just mean that the code inside `DrawerTrigger` could be a own component that we keep in this file